### DISCUSSION TITLE
Component method unrecognized in production

### DISCUSSION THREAD

**Discussion by kwas0ng** [laravel/framework#39321] [Q&A]
**Description**:
I am new to Laravel and I am deploying my website and is almost finished! The only problem is that I keep getting an undefined variable error in production but it works normally in local.

This is the code of the `CPU' component class.  

![image](https://user-images.githubusercontent.com/61017687/138507141-1df52814-0c4e-4771-868b-695cb4200f58.png)

the `setID()` method is being invoked in the blade file and looks like this.

![image](https://user-images.githubusercontent.com/61017687/138507303-33b4a8da-4132-4b53-8cf6-7a72f9c509db.png)

It perfectly works locally and in the live server, it throws an error like this.
![image](https://user-images.githubusercontent.com/61017687/138507431-ea1c3a1d-47ab-438d-99a7-2690eb9af55c.png)

**Comment by adityadees** [comment]
please check `setID` . because `Case Sensitive` might be an issue
